[TASK] Rewrite TypoScript chapter (#2818)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/SymfonyExpressionLanguage/Index.rst

Symfony expression language
===========================

Symfony expression language (SEL) is used by TYPO3 in a couple of places. The most

> well know one are :ref:`TypoScript conditions <typoscript-syntax-global-condition>`.

Reviewer: ```suggestion
well known ones are :ref:`TypoScript conditions <typoscript-syntax-global-condition>`.
```

> well known ones are :ref:`TypoScript conditions <typoscript-syntax-global-condition>`.

Reviewer 2: ```suggestion
well-known ones are :ref:`TypoScript conditions <typoscript-syntax-global-condition>`.
```

> The :ref:`TSref <t3tsref:conditions>` and :ref:`TSconfig <t3tsconfig:conditions>` list

Reviewer 2: ```suggestion
The :ref:`TypoScript <t3tsref:conditions>` and :ref:`TSconfig <t3tsconfig:conditions>` references list
```

> available variables and functions of these contexts. But the TYPO3 core API allows

Reviewer 2: ```suggestion
available variables and functions of these contexts. But the TYPO3 Core API allows
```

enriching expressions with additional functionality, which is what this chapter is about.

Main API
========

> The TYPO3 core API provides a relatively slim API in front of symfony expression

Reviewer 2: ```suggestion
The TYPO3 Core API provides a relatively slim API in front of symfony expression
```

> The TYPO3 Core API provides a relatively slim API in front of symfony expression

Reviewer 2: ```suggestion
The TYPO3 Core API provides a relatively slim API in front of Symfony expression
```

language: Symfony expressions are used in different contexts (TypoScript conditions,
the EXT:form framework, maybe more).

> Class :php:`TYPO3\CMS\Core\ExpressionLanguage\Resolver` in used to prepare the

Reviewer: ```suggestion
Class :php:`TYPO3\CMS\Core\ExpressionLanguage\Resolver` is used to prepare the
```

> Class :php:`TYPO3\CMS\Core\ExpressionLanguage\Resolver` is used to prepare the

Reviewer 2: ```suggestion
The class :php:`TYPO3\CMS\Core\ExpressionLanguage\Resolver` is used to prepare the
```

expression language processor based on a given context (identified by a string,
for example "typoscript"), and loads registered available variables and functions
for this context.

The :ref:`System > Configuration <ext_lowlevel:module-configuration>` module

> provides a list of all registered Symfony Expression Language providers.

Reviewer 2: ```suggestion
provides a list of all registered Symfony expression language providers.
```

Evaluation of single expressions is then initiated calling

> :php:`$myResolver->evaluate()`. While TypoScript cast the return value to :php:`bool`,

Reviewer: ```suggestion
:php:`$myResolver->evaluate()`. While TypoScript casts the return value to :php:`bool`,
```

> symfony expression evaluation can potentially return :php:`mixed`.

Reviewer 2: ```suggestion
Symfony expression evaluation can potentially return :php:`mixed`.
```

Additional variables
====================

Additional variables can be provided by the registered provider class.

> In practice, adding additional variables are used rather seldom: To

Reviewer: ```suggestion
In practice, adding additional variables is used rather seldom: To
```

Reviewer 3: several variables, therefore "are" is correct

Reviewer: it's not about the variables, but about the "adding"

Reviewer 2: Personally, I would go with David's change. But maybe @tomwarwick or @sarahmccarthy123 can shed light on us which one is the correct usage.

Reviewer 4: In this context I would say "is" because it's "the adding" that is singular. You could say "additional variables are added rather seldom" then the "are" relates to "additional variables".

Reviewer 2: Thank you for the clarification :-)

       }
   }

> The below (artificial) implementation calls some external URL based on given variables:

Reviewer: ```suggestion
The (artificial) implementation below calls some external URL based on given variables:
```
